src/cmd/cli/command/commands.go

```diff
@@ -371,11 +371,18 @@ var RootCmd = &cobra.Command{
 
 			// Check if the user has agreed to the terms of service and show a prompt if needed
 			if connect.CodeOf(err) == connect.CodeFailedPrecondition {
-				term.Warn(prettyError(err))
-
 				defer trackCmd(nil, "Terms", P{"reason", err})
-				if err = cli.InteractiveAgreeToS(cmd.Context(), client); err != nil {
-					return err
+				if cliClient.TermsAccepted() {
```

Comment:
I'm tempted to think this local state check should be encapsulated within `client.CheckLoginAndToS` so we don't have to check both potential sources of truth later.

Further, since we want to be able to handle the scenario where the user is unauthenticated, but the terms accepted are accepted locally, I think I might extend the the API of `CheckLoginAndToS` to check the remote state as well as the local state, and then return two booleans and an error. Maybe something like this:

`defang/src/pkg/cli/client/grpc.go`:
```go
func (g GrpcClient) CheckLoginAndToS(ctx context.Context) (bool, bool, error) {
	_, err := g.client.CheckToS(ctx, &connect.Request[emptypb.Empty]{})
	authenticated := err == nil
	accepted := err == nil || TermsAccepted()

	if err != nil {
		if connect.CodeOf(err) == connect.CodeUnauthenticated {
			return false, accepted, nil
		} else if connect.CodeOf(err) == connect.CodeFailedPrecondition {
			return true, accepted, nil
		}
	}

	return authenticated, accepted, err
}
```

Is this idiomatic?

Reply:
I ended up doing it in the "cli" package, as opposed to the client package. This way the client is still very close to the grpc contract, with little added magic.